### PR TITLE
[Doc] Added restriction on aggregation names

### PR DESCRIPTION
**Pull request by jillnogold**
[ML-3782](https://jira.iguazeng.com/browse/ML-3782)

### CODE REVIEW

File: docs/feature-store/transformations.md [hunk 1]
```diff
@@ -44,9 +44,51 @@ to the [feature store example](./basic-demo.html).
...
+
+```{admonition} Warning
+You must ensure that your features will not conflict with the automatically generated feature names. For example, 
+when using `add_aggregation()` on a feature X, you may get a genegated feature name of `X_count_1h`. 
+But if your dataset already contains `X_count_1h`, this would result in either unreliable aggregations or errors.
+```
+
+If either the pattern or the condition is not met, the feature is treated as a static (or "regular") feature.
+    
+These features can be fed into predictive models or czn be used for additional processing and feature generation.
```
Comment (gtopper) [inline docs/feature-store/transformations.md:65]:
```suggestion
These features can be fed into predictive models or can be used for additional processing and feature generation.
```

File: docs/feature-store/transformations.md [hunk 2]
```diff
@@ -44,9 +44,51 @@ to the [feature store example](./basic-demo.html).
...
+
+```{admonition} Notes
+- Internally, the graph step that is created to perform these aggregations is named `"Aggregates"`. If more than one
+   aggregation steps are needed, a unique name must be provided to each, using the `state_name` parameter.
```
Comment (gtopper) [inline docs/feature-store/transformations.md:69]:
```
:param state_name: *Deprecated* - use step_name instead
```
```suggestion
   aggregation steps are needed, a unique name must be provided to each, using the `step_name` parameter.
```

File: docs/feature-store/transformations.md [hunk 3]
```diff
@@ -44,9 +44,51 @@ to the [feature store example](./basic-demo.html).
...
+- `count` 
+- `sum`
+- `sqr` (sum of squares)
+- `max`
+- `min`
+- `first`
+- `last`
+- `avg`
+- `stdvar`
+- `stddev`
```
[the remark this answers is not in the document]
Comment (gtopper) [reply docs/feature-store/transformations.md:83]: I see now that this is just moved from one place to another so I guess it's fine to leave as is in the PR if you prefer.

[the remark this answers is not in the document]
Comment (jillnogold) [reply docs/feature-store/transformations.md]: Added explanations

**Review by gtopper — APPROVED** [review]

**Pull request merged**